src/components/DataTable/DataTable.jsx

```diff
@@ -0,0 +1,260 @@
+
+		onSelectAll({ props: this.props, event });
+	},
+
+	handleRowClick(rowIndex, event) {
+		const {
+			data,
+			onRowClick
+		} = this.props;
+
+		const targetTagName = event.target.tagName.toLowerCase();
+		if (targetTagName === 'td' || targetTagName === 'tr') {
+			onRowClick(data[rowIndex], rowIndex, { props: this.props, event });
+		}
+	},
+
+	handleSort(field, event) {
+		const {
+			onSort
+		} = this.props;
+
+		event.stopPropagation();
+		event.preventDefault();
+		event.bubbles = false;
+		onSort(field, {props: this.props, event});
+	},
+
+	render() {
+
+		const {
+			className,
+			data,
+			hasDetails,
+			isSelectable,
+			...passThroughs
+		} = this.props;
+
+
+		const childComponentElements = findAllChildComponents(this.props, [DataTable.Column, DataTable.ColumnGroup]);
+		const flattenedColumns = _.reduce(childComponentElements, (acc, childComponentElement) => {
+			if (childComponentElement.type === DataTable.Column) {
+				return acc.concat([{props: childComponentElement.props, columnGroupProps: null}]);
+			}
+			if (childComponentElement.type === DataTable.ColumnGroup) {
+				return acc.concat(_.map(findAllChildComponents(childComponentElement.props, [DataTable.Column]), (columnChildComponent) => ({ props: columnChildComponent.props, columnGroupProps: childComponentElement.props })));
+			}
+		}, []);
+
+		const hasGroupedColumns = _.some(childComponentElements, (childComponentElement) => childComponentElement.type === DataTable.ColumnGroup);
+		return (
+			<ScrollTable {...passThroughs} className={boundClassNames('&', className)}>
+				<Thead>
+					<Tr>
+						{isSelectable ? (
+							<Th hasCheckbox rowSpan={hasGroupedColumns ? 2 : null}><Checkbox isSelected={_.every(data, 'isSelected')} onSelect={this.handleSelectAll} /></Th>
+						) : null}
+						{_.map(childComponentElements, ({ props, type }, index) => type === DataTable.Column ? (
+							<Th
+								{..._.omit(props, ['field', 'children', 'width', 'title'])}
+								onClick={DataTable.shouldColumnHandleSort(props) ? _.partial(this.handleSort, props.field) : null}
+								style={{
+									width: props.width
+								}}
+								rowSpan={hasGroupedColumns ? 2 : null}
+								key={_.get(props, 'field', index)}
+							>
+								{props.title || props.children}
+							</Th>
+						) : (
+							<Th
+								colSpan={_.size(DataTable.Column.findInChildren(props.children))}
+								{..._.omit(props, ['field', 'children', 'width', 'title'])}
+								key={_.get(props, 'field', index)}
+							>
+								{props.title || props.children}
+							</Th>
+						))}
+					</Tr>
+					{hasGroupedColumns ? (
+						<Tr>
+							{_.reduce(flattenedColumns, (acc, { props: columnProps, columnGroupProps }, index) => acc.concat(_.isNull(columnGroupProps) ? [] : [(
+								<Th
+									{..._.omit(columnProps, ['field', 'children', 'width', 'title'])}
+									onClick={DataTable.shouldColumnHandleSort(columnProps) ? _.partial(this.handleSort, columnProps.field) : null}
```

Comment:
It's not a big deal right for now, but I can imagine partials causing perf problems down the road. If/when we run into issues we should probably drop the `columnProps.field` as a prop on the `Th` and destructure it out of the callback when fired.